### PR TITLE
Energy Drink Yokes

### PR DESCRIPTION
**Pull request by Geevies**
* Added six pack yokes for energy drinks, a storage item for moving a fast amount of energy drinks quickly. Currently adminspawn only.

image

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/modules/reagents/reagent_containers/food/drinks/yoke.dm [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	starts_with = list(/obj/item/reagent_containers/food/drinks/cans/zoraklax = 6)
+
+/obj/item/storage/box/fancy/yoke/zoracthur
+	starts_with = list(/obj/item/reagent_containers/food/drinks/cans/zoracthur = 6)
+
+/obj/item/storage/box/fancy/yoke/random
+	starts_with = list()
+
+/obj/item/storage/box/fancy/yoke/random/fill()
+	for(var/i = 1; i <= 6, i++)
```
Comment (NonQueueingMatt) [inline code/modules/reagents/reagent_containers/food/drinks/yoke.dm:100]: for the future, these loops are slower than using byond's (aka `for(var/i = 1 to 6)`

**Review by NonQueueingMatt — APPROVED** [review]

**Review by Alberyk — APPROVED** [review]

File: code/modules/reagents/reagent_containers/food/drinks/yoke.dm [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+/obj/item/storage/box/fancy/yoke/MouseDrop(mob/user)
+	if(user != usr || !ishuman(user) || use_check_and_message(user) || user.get_active_hand())
+		return
+	var/mob/living/carbon/human/H = user
+	var/obj/item/organ/external/temp = H.organs_by_name[BP_R_HAND]
+	if(H.hand)
+		temp = H.organs_by_name[BP_L_HAND]
+	if(temp && !temp.is_usable())
+		to_chat(user, SPAN_NOTICE("You try to move your [temp.name], but cannot!"))
+		return
```
Comment (alsoandanswer) [inline code/modules/reagents/reagent_containers/food/drinks/yoke.dm:48]:
```suggestion
	if(user != usr || !ishuman(user) || use_check_and_message(user) || user.get_active_hand())
		return
	var/mob/living/carbon/human/H = user
	var/obj/item/organ/external/temp = H.organs_by_name[BP_R_HAND]
	if(H.hand)
		temp = H.organs_by_name[BP_L_HAND]
	if(temp && !temp.is_usable())
		to_chat(user, SPAN_NOTICE("You try to move your [temp.name], but cannot!"))
		return
```
if(use_check_and_message(usr))
		return 0

File: code/modules/reagents/reagent_containers/food/drinks/yoke.dm [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+		return ..()
+
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		var/obj/item/organ/external/temp = H.organs_by_name[BP_R_HAND]
+		if(H.hand)
+			temp = H.organs_by_name[BP_L_HAND]
+		if(temp && !temp.is_usable())
+			to_chat(user, SPAN_NOTICE("You try to move your [temp.name], but cannot!"))
+			return
```
Comment (alsoandanswer) [inline code/modules/reagents/reagent_containers/food/drinks/yoke.dm:64]:
```suggestion
	if(use_check_and_message(usr))
		return 0
```

**Pull request merged**